### PR TITLE
Run Error api actions

### PR DESCRIPTION
**Pull request by Pralish**
Addresses: https://github.com/restarone/violet_rails/issues/985

Addresses: https://github.com/restarone/violet_rails/issues/1071

### Demo:

https://user-images.githubusercontent.com/50227291/188673867-186d03e9-a8ec-4cb7-adb0-6a6483ba1cec.mov

Since api resource is not saved due to any error, api action will keep track of user input with proper error message under 

`additional_data['api_resource']['properties']` and `additional_data['api_resource']['errors']`

Demo shows error message and user input being accessed by api_action

### Demo fix dead set functionality

https://user-images.githubusercontent.com/50227291/189976658-e7c6fbe8-1f69-4740-9e98-98835a8fae26.mov

### CODE REVIEW

File: app/models/concerns/dynamic_attribute.rb [hunk 1]
```diff
@@ -22,7 +22,8 @@ def parse_dynamic_attribute(value)
         end
 
         # parse erb
-        parser = ERB.new(CGI.unescapeHTML(parsed_text))
+        # unescape html and decode uri-encoded charcters
```
Comment (donrestarone) [inline app/models/concerns/dynamic_attribute.rb:25]: @Pralish this change is not supposed to be here (referenced: https://github.com/restarone/violet_rails/pull/1085)

**Comment by donrestarone** [comment]
@Pralish does this PR also address https://github.com/restarone/violet_rails/issues/1071 ? It was mentioned in the original task since you'd touch code related to it

File: app/models/concerns/dynamic_attribute.rb (hunk 1 above)
Comment (Pralish) [reply app/models/concerns/dynamic_attribute.rb]: @donrestarone Thanks for noticing. I must have checked out from the referenced branch instead of master.

**Comment by Pralish** [comment]
> @Pralish does this PR also address #1071 ? It was mentioned in the original task since you'd touch code related to it

@donrestarone , This PR doesn't. I have looked into the issue and was planning to work on a separate PR.

**Comment by donrestarone** [comment]
@Pralish sounds good. I'll assign the task to you to pick up later

File: db/migrate/20220906063031_add_additional_data_to_api_actions.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddAdditionalDataToApiActions < ActiveRecord::Migration[6.1]
+  def change
+    add_column :api_actions, :additional_data, :jsonb, default: {}
```
Comment (donrestarone) [inline db/migrate/20220906063031_add_additional_data_to_api_actions.rb:3]: rename to `meta_data`

**Comment by donrestarone** [comment]
@Pralish can you address the feedback here? https://github.com/restarone/violet_rails/pull/1086#discussion_r972509593 -- after that we can deploy it for testing!

**Comment by Pralish** [comment]
@donrestarone , renamed the attribute.

I also found an issue. We run error actions incase of any errors. Since we run api actions in background, incase of errors, Sidekiq reruns the job. Error action gets triggered each time the job fails.  We have retries count of 3, so if the job fails all three times , error actions get executed 3 time. 

Shouldn't error action run only after final retry?

**Comment by donrestarone** [comment]
@Pralish if it's a small change yeah let's do it. If not, let's leave as is-- since we are currently not even able to re-run failing jobs

**Pull request merged**